File: 225.ts
class MyStack {

    q: Array<number>

    constructor() {
        this.q = [];
    }

    push(x: number): void {
        this.q.push(x);
        for(let i = 0; i < this.q.length - 1; i++) 
        {
            let num = this.q.shift();
            if (num !== undefined) 
            {
                this.q.push(num);
            }
        }
    }

    pop(): (number | undefined) {
        if(this.q.length > 0)
        {
            return this.q.shift();
        }
        else
        {
            return undefined;
        }
    }

    top(): (number | undefined) {
        if(this.q.length > 0)
        {
            return this.q[0];
        }
        else
        {
            return undefined;
        }
    }

    empty(): boolean {
        return this.q.length === 0;
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * var obj = new MyStack()
 * obj.push(x)
 * var param_2 = obj.pop()
 * var param_3 = obj.top()
 * var param_4 = obj.empty()
 */

File: 225.py
from collections import deque

class MyStack:

    def __init__(self):
        self.d = deque()
        
    def push(self, x: int) -> None:
        self.d.append(x)
        for i in range(0, len(self.d) - 1):
            self.d.append(self.d.popleft())

    def pop(self) -> int:
        return self.d.popleft()        

    def top(self) -> int:
        return self.d[0] if len(self.d) != 0 else None        

    def empty(self) -> bool:
        return True if len(self.d) == 0 else False
        


# Your MyStack object will be instantiated and called as such:
if __name__ == '__main__':
    obj = MyStack()
    obj.push(5)
    obj.push(3)
    obj.push(1)
    print(obj.pop())
    print(obj.top())
    print(obj.empty())